fvm/evm/emulator/emulator.go

```diff
@@ -99,16 +105,22 @@ func (bl *BlockView) DirectCall(call *types.DirectCall) (*types.Result, error) {
 	if err != nil {
 		return nil, err
 	}
-	var res *types.Result
 	switch call.SubType {
 	case types.DepositCallSubType:
-		res, err = proc.mintTo(call.To, call.Value)
+		return proc.mintTo(call.To, call.Value)
 	case types.WithdrawCallSubType:
-		res, err = proc.withdrawFrom(call.From, call.Value)
+		return proc.withdrawFrom(call.From, call.Value)
+	case types.DeployCallSubType:
+		if call.HasNonEmptyToField() {
+			return proc.deployAt(call.From, call.To, call.Data, call.GasLimit, call.Value)
+		}
+		fallthrough
 	default:
-		res, err = proc.run(call.Message(), types.DirectCallTxType)
+		// set message nonce (needed for deploy)
+		msg := call.Message()
+		msg.Nonce = proc.state.GetNonce(call.From.ToCommon())
```

Comment:
couldn't setting the nonce here be part of the `proc.run`?

Reply:
yeah good idea, moved it there, not sure why I put it here in the first place